quantum/painter/lvgl/qp_lvgl.c

```diff
@@ -48,67 +54,85 @@ static uint32_t tick_task_callback(uint32_t trigger_time, void *cb_arg) {
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// Quantum Painter LVGL Integration API: qp_lvgl_start
+// Quantum Painter LVGL Integration API: qp_lvgl_attach
 
-bool qp_lvgl_start(painter_device_t device) {
+bool qp_lvgl_attach(painter_device_t device) {
     qp_dprintf("qp_lvgl_start: entry\n");
+    qp_lvgl_detach();
 
     struct painter_driver_t *driver = (struct painter_driver_t *)device;
     if (!driver->validate_ok) {
-        qp_dprintf("qp_lvgl_start: fail (validation_ok == false)\n");
+        qp_dprintf("qp_lvgl_attach: fail (validation_ok == false)\n");
+        qp_lvgl_detach();
         return false;
     }
 
-    selected_display = device;
-
     // Setting up the tasks
     lvgl_state_t *lv_tick_inc_state = &lvgl_states[0];
-    lv_tick_inc_state->fnc_id = 0;
-    lv_tick_inc_state->delay_ms = 1;
-    lv_tick_inc_state->defer_token = defer_exec_advanced(lvgl_executors, 2, 1, tick_task_callback, lv_tick_inc_state);
+    lv_tick_inc_state->fnc_id       = 0;
+    lv_tick_inc_state->delay_ms     = 1;
+    lv_tick_inc_state->defer_token  = defer_exec_advanced(lvgl_executors, 2, 1, tick_task_callback, lv_tick_inc_state);
 
     if (lv_tick_inc_state->defer_token == INVALID_DEFERRED_TOKEN) {
-        qp_dprintf("qp_lvgl_start_tasks: fail (could not set up qp_lvgl executor)\n");
+        qp_dprintf("qp_lvgl_attach: fail (could not set up qp_lvgl executor)\n");
+        qp_lvgl_detach();
         return false;
     }
 
     lvgl_state_t *lv_task_handler_state = &lvgl_states[1];
-    lv_task_handler_state->fnc_id = 1;
-    lv_task_handler_state->delay_ms = 5;
-    lv_task_handler_state->defer_token = defer_exec_advanced(lvgl_executors, 2, 5, tick_task_callback, lv_task_handler_state);
+    lv_task_handler_state->fnc_id       = 1;
+    lv_task_handler_state->delay_ms     = 5;
+    lv_task_handler_state->defer_token  = defer_exec_advanced(lvgl_executors, 2, 5, tick_task_callback, lv_task_handler_state);
 
     if (lv_task_handler_state->defer_token == INVALID_DEFERRED_TOKEN) {
-        qp_dprintf("qp_lvgl_start_tasks: fail (could not set up qp_lvgl executor)\n");
+        qp_dprintf("qp_lvgl_attach: fail (could not set up qp_lvgl executor)\n");
+        qp_lvgl_detach();
         return false;
     }
 
     // Init LVGL
     lv_init();
 
-    // Setting up lvgl display buffer
+    // Set up lvgl display buffer
     static lv_disp_draw_buf_t draw_buf;
-    static lv_color_t color_buffer[QUANTUM_PAINTER_LVGL_DISPLAY_WIDTH*(QUANTUM_PAINTER_LVGL_DISPLAY_HEIGHT / 10)];                       /*Declare a buffer for 1/10 screen size*/
-    lv_disp_draw_buf_init(&draw_buf, color_buffer, NULL, QUANTUM_PAINTER_LVGL_DISPLAY_WIDTH*(QUANTUM_PAINTER_LVGL_DISPLAY_HEIGHT / 10)); /*Initialize the display buffer.*/
+    // Allocate a buffer for 1/16 screen size
+    const size_t count_required = driver->panel_width * driver->panel_height / 16;
```

Comment:
Is there a reason why 1/16 of the screen size? In my test using a smaller buffer result in bad-performance due to the increase in SPI transactions.

For reference: [ LVGL devs recommend 1/10 of the size ](https://docs.lvgl.io/8.2/porting/display.html#draw-buffer)

Reply:
Was a leftover from testing why I had screen tearing, will revert.